gtk/src/list_store.rs

```diff
@@ -46,31 +45,39 @@ impl<O: IsA<ListStore>> GtkListStoreExtManual for O {
     fn insert_with_values(
         &self,
         position: Option<u32>,
-        columns: &[u32],
-        values: &[&dyn ToValue],
+        columns_and_values: &[(u32, &dyn ToValue)],
     ) -> TreeIter {
         unsafe {
             assert!(position.unwrap_or(0) <= i32::max_value() as u32);
-            assert_eq!(columns.len(), values.len());
             let n_columns = ffi::gtk_tree_model_get_n_columns(
                 self.as_ref().upcast_ref::<TreeModel>().to_glib_none().0,
             ) as u32;
-            assert!(columns.len() <= n_columns as usize);
-            for (&column, value) in columns.iter().zip(values.iter()) {
-                assert!(column < n_columns);
+            assert!(columns_and_values.len() <= n_columns as usize);
```

Comment:
Please add a error message as second argument.

Reply:
You mean the assertion in words? Or what error message?

Reply:
Just the second argument (which will be the error message). Something like "too many columns are present, expected {}, found {}".

Reply:
Ah, that makes sense yes